lombok-modules/lombok-custom/pom.xml

```diff
@@ -9,8 +9,9 @@
 
     <parent>
         <groupId>com.baeldung</groupId>
-        <artifactId>lombok-modules</artifactId>
+        <artifactId>parent-modules</artifactId>
```

Comment:
this is not needed, the lombok-custom can still refer to parent-modules as its parent

Reply:
To apply this, I had to use the maven-compiler-plugin for the submodule (lombok-custome) to override the JDK version and make it compilable.